modules/navier_stokes/include/userobjects/RhieChowMassFlux.h

```diff
@@ -0,0 +1,226 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "GeneralUserObject.h"
+#include "BlockRestrictable.h"
+#include "FaceArgInterface.h"
+#include "CellCenteredMapFunctor.h"
+#include "FaceCenteredMapFunctor.h"
+#include "VectorComponentFunctor.h"
+#include "LinearFVAnisotropicDiffusion.h"
+#include <unordered_map>
+#include <set>
+#include <unordered_set>
+
+#include "libmesh/petsc_vector.h"
+
+class MooseMesh;
+class INSFVVelocityVariable;
+class INSFVPressureVariable;
+namespace libMesh
+{
+class Elem;
+class MeshBase;
+}
+
+/**
+ * User object responsible for determining the face fluxes using the Rhie-Chow interpolation in a
+ * segregated solver that uses the linear FV formulation.
+ */
+class RhieChowMassFlux : public GeneralUserObject,
+                         public BlockRestrictable,
+                         public NonADFunctorInterface,
```

Comment:
~IMO you should inherit from the `ADFunctorInterface`. Then you will get errors if a user passed in a functor with derivatives attached~ I forgot for a second that linear FV variables are still AD functors